packages/types/src/Event.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright 2017-2018 @polkadot/types authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { Constructor, ConstructorDef } from './types';
+
+import { isUndefined, u8aToHex } from '@polkadot/util';
+
+import Base from './codec/Base';
+import Struct from './codec/Struct';
+import Tuple from './codec/Tuple';
+import U8aFixed from './codec/U8aFixed';
+import { TypeDef, getTypeClass, getTypeDef } from './codec/createType';
+import Metadata from './Metadata';
+
+const EventTypes: { [index: string]: Constructor<EventData> } = {};
+
+class EventData<
+  // S & T definitions maps to what we have in Struct (naming documented there)
+  S extends ConstructorDef = { [index: string]: Constructor<Base> },
+  V extends { [K in keyof S]: any } = { [K in keyof S]: any }
+> extends Tuple<S> {
+  private _typeDef: Array<TypeDef>;
+
+  constructor (Types: S, value: V | Array<any> = {} as V, typeDef: Array<TypeDef>) {
+    super(Types, value);
+
+    this._typeDef = typeDef;
+  }
+
+  get typeDef (): Array<TypeDef> {
+    return this._typeDef;
+  }
+
+  toHex (): string {
```

Comment:
Already defined in Struct.

Reply:
Grrr, added it for debugging actually. Thanks.